numpy/ma/core.py

```diff
@@ -6358,7 +6360,7 @@ def tolist(self, fill_value=None):
         inishape = self.shape
         result = np.array(self._data.ravel(), dtype=object)
```

Comment:
I wonder why the `.ravel()` here is necessary? I'd think this would work fine just as,
```
result = np.array(self._data, dtype=object)
result[_mask] = None
return result
```
Indeed, in principle,
```
return self.astype(object).filled(None)
```
should work (sorry, I really should test, but running out of time...)

Reply:
Probably the code can be improved, but I strongly prefer to do it in a separate PR.

Reply:
Yes, fine to here just to the `.reshape` replacement.